src/monitoring_service/cli.py

```diff
@@ -10,57 +11,38 @@
     CONTRACT_USER_DEPOSIT,
 )
 from raiden_contracts.contract_manager import ContractManager, contracts_precompiled_path
-from raiden_libs.cli import (
-    blockchain_options,
-    common_options,
-    connect_to_blockchain,
-    validate_address,
-)
+from raiden_libs.cli import blockchain_options, common_options
 from raiden_libs.contract_info import START_BLOCK_ID
-from raiden_libs.types import Address
 
 log = structlog.get_logger(__name__)
 
 
-CONTEXT_SETTINGS = dict(default_map={'main': {'confirmations': DEFAULT_REQUIRED_CONFIRMATIONS}})
-
-
-@click.command(context_settings=CONTEXT_SETTINGS)
-@click.option(
-    '--monitor-contract-address',
-    type=str,
-    help='Address of the token monitor contract',
-    callback=validate_address,
-)
+@blockchain_options()
+@click.command()
 @click.option(
     '--min-reward',
     default=0,
     type=click.IntRange(min=0),
     help='Minimum reward which is required before processing requests',
 )
+@click.option(
+    '--confirmations',
+    default=DEFAULT_REQUIRED_CONFIRMATIONS,
+    type=click.IntRange(min=0),
+    help='Number of block confirmations to wait for',
+)
 @common_options('raiden-monitoring-service')
-@blockchain_options
 def main(
     private_key: str,
     state_db: str,
-    eth_rpc: str,
-    registry_address: Address,
-    user_deposit_contract_address: Address,
-    start_block: BlockNumber,
-    monitor_contract_address: Address,
+    web3: Web3,
+    contract_infos: dict,
+    confirmations: BlockNumber,
```

Comment:
Not for this PR, but I think we should start to differentiate between absolute `BlockNumber`s and relative `BlockInterval`s. I think there's something similar in Raiden.